cfme/tests/containers/test_static_custom_attributes.py

```diff
@@ -124,3 +158,79 @@ def test_add_date_value_with_wrong_value(provider):
 
     if hasattr(provider.summary, 'custom_attributes'):
         assert 'nondate' not in provider.summary.custom_attributes
+
+
+# CMP-10405
+
+def test_edit_non_exist_attribute(provider):
+    setup_dataset(provider, verify_exists=False)
+
+    ca = choice(ATTRIBUTES_DATASET)
+    try:
+        # Note: we need to implement it inside the test instead of using
+        #       the API (provider.edit_custom_attributes) in order to
+        #       specify the href and yield the exception
+        payload = {
+            "action": "edit",
+            "resources": [{
+                "href": '{}/custom_attributes/9876543210000000'
+                        .format(provider.href()),
+                "value": ca.value
+            }]}
+        provider.appliance.rest_api.post(
+            path.join(provider.href(), 'custom_attributes'), **payload)
```

Comment:
Good switch to pytest.raises - but a note. in general, try to limit what is in the `with()` block to the call that will actually raise the exception. So setup `payload` beforehand and only make your rest_api call in the `with()`